source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -242,7 +242,7 @@ void EnvoyQuicClientConnection::onFileEvent(uint32_t events,
   ASSERT(events & (Event::FileReadyType::Read | Event::FileReadyType::Write));
 
   if (events & Event::FileReadyType::Write) {
-    OnCanWrite();
+    OnBlockedWriterCanWrite();
```

Comment:
does this call still work if the writer is not blocked?

Reply:
oh nevermind, I see that it unblocks the writer before calling OnCanWrite()

Reply:
Yes, `writer_->SetWritable()` is a no-op on unblocked writer.